src/org/omegat/filters4/xml/AbstractXmlFilter.java

```diff
@@ -262,6 +264,9 @@ public void processFile(BufferedReader inReader, BufferedWriter writer, FilterCo
                                 fromEventToWriter(event, strWriter);
                             }
                         } else {
+                            // still in non-event mode, we always write
+                            // exactly what was in the source
+                            fromReaderToWriter(strReader, strWriter);
```

Comment:
Yes I wanted to suggest to move fromReaderToWriter in the else part (rather than having a specific if as in the comment) but apparently you saw this possiblity.

OK for the moment. I would prefer a solution which works for all StaX implementations (in case you decide later for a new change...) but I will think about this later.  You can merge like this now.

Reply:
> OK for the moment. I would prefer a solution which works for all StaX implementations (in case you decide later for a new change...) but I will think about this later. You can merge like this now.

Good news!   >> **works for all StaX implementations**

Reply:
No, I tested with OmegaT 5.7 + plugin, the <body> completely disappears in this case